src/core/BufferGeometry.js

```diff
@@ -1148,15 +1148,10 @@ BufferGeometry.merge = function( geometries ) {
 		geom2 = geometries[ m ];
 		if ( geom2.index ) {
 
-			geom2 = BufferGeometry.splitVertices( geom2 );
+			geom2 = geom2.toNonIndexed();
 			geometries[ m ] = geom2;
 
 		}
-		if ( ! geom2.attributes.normal ) {
-
-			// geometries[ m ].geometry = computeNormals( geom2 );
```

Comment:
Not sure if anyone is interested, but I also have a function that computes normal vectors that are sharp for corners, and smooth for flat regions. It uses an angle threshold to determine whether the surface is smooth or sharp. I left it out of this pull request.